File: react_native_app/src/features/productos/domain/usecases/deleteProducto.ts
import { UseCase } from '../../../../core/usecase/usecase';
import { ProductoRepository } from '../repositories/ProductoRepository';

/** Elimina un producto (DELETE a la API) y lo quita del estado en memoria. */
export class DeleteProducto implements UseCase<void, number> {
  constructor(private readonly repository: ProductoRepository) {}

  call(id: number): Promise<void> {
    return this.repository.deleteProducto(id);
  }
}

File: flutter_app/lib/features/productos/domain/usecases/delete_producto.dart
import '../../../../core/usecase/usecase.dart';
import '../repositories/producto_repository.dart';

/// Elimina un producto (DELETE a la API) y lo quita del estado en memoria.
class DeleteProducto implements UseCase<void, int> {
  final ProductoRepository repository;
  DeleteProducto(this.repository);

  @override
  Future<void> call(int id) => repository.deleteProducto(id);
}
